### PR TITLE
Room Details + Profile Header 

### PR DESCRIPTION
**Pull request by rauchp**
- Made Room Details reload more efficiently to reduce stale data, and added Activity Indicator to its ScrollView. 
  - I think adding reload indicators that react to data updates in roomDetails. I think these throughout are app are crucial now. When adding/removing data on screen if it's not updated on a reload, it looks janky and unpolished. 
- Profile Header Fix (important)
- Changing profile from header now triggers room details reload.

### CODE REVIEW

File: civilapp/src/discover/DiscoverScreen.js [hunk 1]
```diff
@@ -91,7 +90,7 @@ class DiscoverScreen extends PureComponent {
       <View style={styles.headerView}>
         <View style={{ flex: 0.225, flexDirection: 'row', justifyContent: 'center' }}>
           <TouchableOpacity style={styles.headerButton} onPress={this.openDrawer}>
-            <Image style={{ height: 30, width: 30 }} resizeMode='contain' source={this.profileImage} />
+            <Image style={{ height: 30, width: 30 }} resizeMode='contain' source={this.state.profileImage} />
```
Comment (EphemeralX) [inline civilapp/src/discover/DiscoverScreen.js]: nit: deconstruct

File: civilapp/src/current/UserRow.js [hunk 2]
```diff
@@ -26,6 +26,7 @@ class UserRow extends React.Component {
         onPress={() => {
           this.props.navigation.navigate('ProfilePrimary', {
             userId: this.props.userId,
+            refreshUserInfo: this.props.refreshUserInfo
```
Comment (EphemeralX) [inline civilapp/src/current/UserRow.js]: nit: deconstruct

File: civilapp/src/current/RoomDetails.js [hunk 3]
```diff
@@ -150,52 +154,25 @@ class RoomDetails extends React.Component {
...
+    })
+  }
+
+  componentDidUpdate(prevProps) {
     if (this.props.navigation.getParam('shouldUpdateHeader')) {
-      refreshScreen()
+      this.refreshScreen()
       this.props.navigation.setParams({ shouldUpdateHeader: false })
     }
     if (!prevProps.navigation.state.isDrawerOpen && this.props.navigation.state.isDrawerOpen) {
```
Comment (EphemeralX) [inline civilapp/src/current/RoomDetails.js]: nit: deconstruct

File: civilapp/src/browse/BrowseScreen.js [hunk 4]
```diff
@@ -102,7 +101,7 @@ class BrowseScreen extends PureComponent {
       <View style={styles.headerView}>
         <View style={styles.headerButtonView}>
           <TouchableOpacity style={styles.headerButton} onPress={this.openDrawer}>
-            <Image style={{ height: 30, width: 30 }} resizeMode='contain' source={this.profileImage} />
+            <Image style={{ height: 30, width: 30 }} resizeMode='contain' source={this.state.profileImage} />
```
Comment (EphemeralX) [inline civilapp/src/browse/BrowseScreen.js]: nit: deconstruct

**Comment by rauchp** [comment]
sorry for the lack of deconstructing.

**Pull request merged**